tools/scripts/features/frcnn/extract_features_frcnn.py

```diff
@@ -0,0 +1,296 @@
+                or single_features["attr_probs"][idx] < confidence
+            ):
+                single_features["obj_ids"] = torch.cat(
+                    [
+                        single_features["obj_ids"][0:idx],
+                        single_features["obj_ids"][idx + 1 :],
+                    ]
+                )
+                single_features["obj_probs"] = torch.cat(
+                    [
+                        single_features["obj_probs"][0:idx],
+                        single_features["obj_probs"][idx + 1 :],
+                    ]
+                )
+                single_features["attr_ids"] = torch.cat(
+                    [
+                        single_features["attr_ids"][0:idx],
+                        single_features["attr_ids"][idx + 1 :],
+                    ]
+                )
+                single_features["attr_probs"] = torch.cat(
+                    [
+                        single_features["attr_probs"][0:idx],
+                        single_features["attr_probs"][idx + 1 :],
+                    ]
+                )
+                single_features["boxes"] = torch.cat(
+                    [
+                        single_features["boxes"][0:idx, :],
+                        single_features["boxes"][idx + 1 :, :],
+                    ]
+                )
+                single_features["preds_per_image"] = (
+                    single_features["preds_per_image"] - 1
+                )
+                single_features["roi_features"] = torch.cat(
+                    [
+                        single_features["roi_features"][0:idx, :],
+                        single_features["roi_features"][idx + 1 :, :],
+                    ]
+                )
+                single_features["normalized_boxes"] = torch.cat(
+                    [
+                        single_features["normalized_boxes"][0:idx, :],
+                        single_features["normalized_boxes"][idx + 1 :, :],
+                    ]
+                )
+                removed = True
+            if not removed:
+                idx += 1
+
+        feat_list = single_features["roi_features"]
+
+        boxes = single_features["boxes"][: self.args.num_features].cpu().numpy()
+        num_boxes = self.args.num_features
+        objects = single_features["obj_ids"][: self.args.num_features].cpu().numpy()
+        probs = single_features["obj_probs"][: self.args.num_features].cpu().numpy()
+        width = single_features["sizes"][1].item()
+        height = single_features["sizes"][0].item()
+        info_list = {
+            "bbox": boxes,
+            "num_boxes": num_boxes,
+            "objects": objects,
+            "cls_prob": probs,
+            "image_width": width,
+            "image_height": height,
+        }
+
+        return single_features, feat_list, info_list
+
+    def _chunks(self, array, chunk_size):
+        for i in range(0, len(array), chunk_size):
+            yield array[i : i + chunk_size], i
```

Comment:
this loop accessing in the array like this makes me feel like it is going to error (index out of bounds). :)

Reply:
What is actually nice about this is because we are using an array range, you won't get an index out of bounds error. I could be misunderstanding what you're saying though, in which case I can change this and also the `extract_features_vmb.py` version.

For example, if our array is [1, 2, 3] and I run `print(array[4])` it will print out of bounds, but if I run `print[2:4]` it will run successfully.

Other option is I could create a test file for this method? Just a thought